src/pages/CustomerInvoicesPage.js

```diff
@@ -3,252 +3,231 @@
-      isAscending: false,
-    };
-  }
-
-  onNewInvoice = otherParty => {
-    const { database, currentUser } = this.props;
-    let invoice;
-    database.write(() => {
-      invoice = createRecord(database, 'CustomerInvoice', otherParty, currentUser);
-    });
-    this.navigateToInvoice(invoice);
+import {
+  PageButton,
+  CheckedComponent,
+  UncheckedComponent,
+  DisabledCheckedComponent,
+  DisabledUncheckedComponent,
+} from '../widgets';
+import { DataTable, Row, Cell, CheckableCell, DataTableHeaderRow } from '../widgets/DataTable';
+import {
+  selectRow,
+  deselectRow,
+  deselectAll,
+  sortData,
+  filterData,
+  openBasicModal,
+  closeBasicModal,
+  deleteTransactionsById,
+} from './dataTableUtilities/actions';
+import { newSortDataBy } from '../utilities';
+import usePageReducer from '../hooks/usePageReducer';
+import DataTablePageView from './containers/DataTablePageView';
+
+import { SUSSOL_ORANGE, newDataTableStyles, newPageStyles } from '../globalStyles';
+
+const keyExtractor = item => item.id;
+
+const initialState = () => {
+  const backingData = UIDatabase.objects('CustomerInvoice');
+  return {
+    backingData,
+    data: newSortDataBy(backingData.slice(), 'serialNumber', false),
+    keyExtractor,
+    dataState: new Map(),
+    searchTerm: '',
+    filterDataKeys: ['otherParty.name'],
+    sortBy: 'serialNumber',
+    isAscending: false,
+    modalKey: '',
+    hasSelection: false,
   };
+};
 
-  onDeleteConfirm = () => {
-    const { selection, transactions } = this.state;
-    const { database } = this.props;
-
-    database.write(() => {
-      const transactionsToDelete = [];
-      for (let i = 0; i < selection.length; i += 1) {
-        const transaction = transactions.find(
-          currentTransaction => currentTransaction.id === selection[i]
-        );
-        if (transaction.isValid() && !transaction.isFinalised) {
-          transactionsToDelete.push(transaction);
-        }
-      }
-      database.delete('Transaction', transactionsToDelete);
-    });
-    this.setState({ selection: [] });
-    this.refreshData();
-  };
-
-  onDeleteCancel = () => {
-    this.setState({ selection: [] });
-    this.refreshData();
-  };
-
-  onRowPress = invoice => this.navigateToInvoice(invoice);
-
-  onSelectionChange = newSelection => this.setState({ selection: newSelection });
+export const CustomerInvoicesPage = ({ currentUser, navigateTo, routeName }) => {
+  const [state, dispatch, instantDebouncedDispatch, debouncedDispatch] = usePageReducer(
+    routeName,
+    initialState()
+  );
+  const { data, dataState, sortBy, isAscending, columns, modalKey, hasSelection } = state;
 
-  navigateToInvoice = invoice => {
-    const { database, navigateTo } = this.props;
+  const renderNewInvoiceButton = () => (
+    <PageButton
+      text={buttonStrings.new_invoice}
+      onPress={() => dispatch(openBasicModal('Anything will do'))}
+    />
+  );
 
+  const navigateToInvoice = invoice => {
```

Comment:
Just checking in on using redux navigation - are we to do that after?

Reply:
Yea, I'd rather you show me the light on that one after.

Reply:
cool~